bridges/snowbridge/primitives/outbound-queue/src/v2/converter/convert.rs

```diff
@@ -109,18 +111,105 @@ where
+				Asset { id: AssetId(inner_location), fun: Fungible(amount) } =>
+					match inner_location.unpack() {
+						(0, [AccountKey20 { network, key }]) if self.network_matches(network) =>
+							Some((H160(*key), *amount)),
+						// To allow ether
+						(0, []) => Some((H160([0; 20]), *amount)),
+						_ => None,
+					},
+				_ => None,
+			}
+			.ok_or(AssetResolutionFailed)?;
+
+			// transfer amount must be greater than 0.
+			ensure!(amount > 0, ZeroAssetTransfer);
+
+			commands.push(Command::UnlockNativeToken { token, recipient, amount });
+		}
+		Ok(commands)
+	}
+
+	/// Extract polkadot native assets
+	fn extract_polkadot_native_assets(
+		&mut self,
+		pnas: &Assets,
+		deposit_assets: &AssetFilter,
+		recipient: H160,
+	) -> Result<Vec<Command>, XcmConverterError> {
+		let mut commands: Vec<Command> = Vec::new();
+		ensure!(pnas.len() > 0, NoReserveAssets);
+		for pna in pnas.clone().inner().iter() {
+			if !deposit_assets.matches(pna) {
+				return Err(FilterDoesNotConsumeAllAssets);
+			}
+
+			// Only fungible is allowed
+			let (asset_id, amount) = match pna {
+				Asset { id: AssetId(inner_location), fun: Fungible(amount) } =>
+					Some((inner_location.clone(), *amount)),
+				_ => None,
+			}
+			.ok_or(AssetResolutionFailed)?;
+
+			// transfer amount must be greater than 0.
+			ensure!(amount > 0, ZeroAssetTransfer);
+
+			// Ensure PNA already registered
+			let token_id = TokenIdOf::convert_location(&asset_id).ok_or(InvalidAsset)?;
+			let expected_asset_id = ConvertAssetId::convert(&token_id).ok_or(InvalidAsset)?;
+			ensure!(asset_id == expected_asset_id, InvalidAsset);
+
+			commands.push(Command::MintForeignToken { token_id, recipient, amount });
+		}
+		Ok(commands)
+	}
 
+	/// Convert the xcm into an outbound message which can be dispatched to
+	/// the Gateway contract on Ethereum
+	///
+	/// Assets being transferred can either be Polkadot-native assets (PNA)
+	/// or Ethereum-native assets (ENA).
+	///
+	/// Expected Input Syntax:
+	/// ```ignore
+	/// WithdrawAsset(ETH)
+	/// PayFees(ETH)
+	/// ReserveAssetDeposited(PNA) | WithdrawAsset(ENA)
+	/// AliasOrigin(Origin)
+	/// DepositAsset(Asset)
+	/// Transact() [OPTIONAL]
+	/// SetTopic(Topic)
+	/// ```
+	/// Notes:
+	/// a. Fee asset will be checked and currently only Ether is allowed
+	/// b. For a specific transfer, either `ReserveAssetDeposited` or `WithdrawAsset` should be
+	/// 	present
+	/// c. `ReserveAssetDeposited` and `WithdrawAsset` can also be present in any order within the
+	/// 	same message
+	/// d. Currently, teleport asset is not allowed, transfer types other than
+	/// 	above will cause the conversion to fail
+	/// e. Currently, `AliasOrigin` is always required, can distinguish the V2 process from V1.
+	/// 	it's required also for dispatching transact from that specific origin.
+	/// f. SetTopic is required for tracing the message all the way along.
+	fn to_ethereum_message(&mut self) -> Result<Message, XcmConverterError> {
```

Comment:
Just call this `convert`

```suggestion
	fn convert(&mut self) -> Result<Message, XcmConverterError> {
```

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/acbb04bbed665ecca0c6a6fbb21ed4f1c8d93cba